src/database/database-thread.c

```diff
@@ -116,8 +118,14 @@ void *DB_thread(void *val)
 
 		BREAK_IF_KILLED();
 
-		// Sleep 1 sec
-		thread_sleepms(DB, 1000);
+		// Inspect setupVars.conf to see if Pi-hole blocking is enabled
+		if(get_and_clear_event(RELOAD_BLOCKINGSTATUS))
+			check_blocking_status();
+
+		BREAK_IF_KILLED();
+
+		// Sleep 0.1 sec
+		thread_sleepms(DB, 100);
```

Comment:
Why do we need a sleep here?

Reply:
There was a sleep here before, too (see deleted line 120). Without a sleep, this thread will start spinning at 100% checking the flags millions of times every second. No need to waste CPU resources on this bit so we say it's safe for the thread to sleep for 100 msec until it should recheck all the flags again.